packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1632,6 +1684,18 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
         }
         : undefined,
       advancedOptions: props.advancedOptions,
+      offPeakWindowOptions: props.offPeakWindow ? {
+        enabled: true,
+        offPeakWindow: {
+          windowStartTime: props.offPeakWindow.windowStartTime ?? {
+            hours: 22,
+            minutes: 0,
+          },
+        },
+      } : undefined,
```

Comment:
so the reason why I'm not in love with this API yet is because to specify offPeakWindow with the given defaults we have to do:

```ts
offPeakWindow: {},
```

That's not very intuitive. Which now makes me understand why `enabled` was a prop in the first place :). Still, though, this potential experience to get the default window isn't great either:

```ts
offPeakWindow: {
  enabled: true,
},
```

It feels weird to supply an object with just a boolean prop. So I suggest we do the following:

```ts
offPeakWindowEnabled: true,
// if people want to customize the window
// offPeakWindowStart: {
//  hours: 22,
//  minutes: 0,
// },
```

I think having two separate properties on the base construct is the way to go and what we've generally done in the past for similar situations.

Of course, the default for `offPeakWindowEnabled` is `true if offPeakWindowStart is set, false otherwise`.